fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/LogicalPartitionTopNToPhysicalPartitionTopN.java

```diff
@@ -99,6 +107,66 @@ private List<PhysicalPartitionTopN<? extends Plan>> generatePhysicalPartitionTop
         }
     }
 
+    /**
+     * check if partition keys' ndv is almost near the total row count.
+     * if yes, it is not suitable for two phase global partition topn.
+     */
+    private boolean checkTwoPhaseGlobalPartitionTopn(LogicalPartitionTopN<? extends Plan> logicalPartitionTopN) {
+        logicalPartitionTopN.getGroupExpression().get().getOwnerGroup().getStatistics();
+        int globalPartitionTopnThreshold = ConnectContext.get().getSessionVariable().getGlobalPartitionTopNThreshold();
+        if (logicalPartitionTopN.getGroupExpression().isPresent()) {
+            Group group = logicalPartitionTopN.getGroupExpression().get().getOwnerGroup();
+            if (group != null && group.getStatistics() != null) {
+                Statistics stats = group.getStatistics();
+                double rowCount = stats.getRowCount();
+                List<Expression> partitionKeys = logicalPartitionTopN.getPartitionKeys();
+                if (!checkPartitionKeys(partitionKeys)) {
+                    return false;
+                }
+                List<ColumnStatistic> partitionByKeyStats = partitionKeys.stream()
+                        .map(partitionKey -> {
+                            ColumnStatistic partitionKeyStats = stats.findColumnStatistics(partitionKey);
+                            if (partitionKeyStats == null || partitionKeyStats.isUnKnown) {
```

Comment:
and we can use filter instead of null check with if.
   .filter(Objects::nonNull)
   .filter(stat -> !stat.isUnKnown)